custom/enikshay/case_utils.py

```diff
@@ -103,32 +102,38 @@ def get_person_case_from_episode(domain, episode_case_id):
     )
 
 
-def get_open_occurrence_case_from_person(domain, person_case_id):
+def get_open_occurrence_case_from_person(domain, person_case_id, last_closed=False):
```

Comment:
do you think it would make sense to make a different version of this function called `get_any_occurrence_from_person` or something similar? It's a bit confusing to have this be `get_open...` with the option to return closed cases too.

Reply:
hmm..i just started to make that change and making two different methods was becoming costly.
For ex
```
def get_open_or_first_closed_occurence_case_from_person(domain, person_case_id):
    open_occurence_case = get_open_occurrence_case_from_person(domain, person_case_id)
    if not open_occurence_case:
        case_accessor = CaseAccessors(domain)
        occurrence_cases = case_accessor.get_reverse_indexed_cases([person_case_id])
        all_occurrence_cases = [case for case in occurrence_cases if case.type == CASE_TYPE_OCCURRENCE]
        if all_occurrence_cases:
            all_occurrence_cases.sort(key=lambda x: x.closed_on, reverse=True)
            return all_occurrence_cases[0]
```
Would need to fetch reverse indexes twice in case there is no open case. Plus the method call would get a little difficult [here](https://github.com/dimagi/commcare-hq/pull/17861/files#diff-f9aa24b679e7af67fd33a5d7a75f1485R187).

I think just renaming this method itself would be fine. Renaming this to `get_occurrence_case_from_person` with that argument defaulting to False should be fine i guess.

Reply:
Made the rename change for now.